kafka/consumer_msg.c

```diff
@@ -66,6 +66,31 @@ lua_consumer_msg_value(struct lua_State *L) {
     return 1;
 }
 
+int
+lua_consumer_msg_headers(struct lua_State *L) {
+    msg_t *msg = lua_check_consumer_msg(L, 1);
+    if (msg->headers == NULL)
+        return 0;
+
+    lua_newtable(L);
+
+    size_t idx = 0;
+    const char *key;
+    const void *val;
+    size_t size;
+
+    while (!rd_kafka_header_get_all(msg->headers, idx++,
+                                    &key, &val, &size)) {
+        lua_pushstring(L, key);
+        if (val != NULL)
+            lua_pushlstring(L, val, size);
+        else
+            lua_pushliteral(L, "NULL"); // replace with box.NULL ?
```

Comment:
I think it should be replaced with box.NULL, since theoretically the string "NULL" can be in a real value

Reply:
Fixed